lib/Entity/Schedule.php

```diff
@@ -474,6 +474,37 @@ public function validate()
         // Check recurrenceDetail every is positive
         if ($this->recurrenceType != '' && ($this->recurrenceDetail === null || $this->recurrenceDetail <= 0))
             throw new InvalidArgumentException(__('Repeat every must be a positive number'), 'recurrenceDetail');
+        
+        // Check and disallow Minute/Hourly repeats to be indefinite (or too long)
+        $twelveHoursInSeconds = 12 * 60 * 60;
+        $oneWeekInSeconds = 24 * 7 * 60 * 60;
+        if($this->recurrenceType == 'Minute') {
+
+            if (empty($this->recurrenceRange)) {
+                throw new InvalidArgumentException(__('You must provide a date and time to end the Repeat'), 'recurrenceRange');
+            }
+
+            $distance = ($this->getDate()->parse($this->recurrenceRange, 'U')->diffInSeconds($this->getDate()->parse($this->fromDt, 'U'))) / $this->recurrenceDetail;
+
+            if ($distance > $twelveHoursInSeconds) {
+                // Recurrence range cannot be more than 12 hours
+                $exceedLimit = $this->getDate()->parse($this->fromDt, 'U')->addSeconds($twelveHoursInSeconds * $this->recurrenceDetail );
+                throw new InvalidArgumentException(__('The date and time to end the Repeat for this Event cannot exceed '.$exceedLimit), 'recurrenceRange');
```

Comment:
For translated strings that have a variable, we should use sprintf so that the translator can put the variable in the appropriate place. E.g.

```
sprintf(__('The end time for this event can only be %d seconds in the future because of the repeating interval being minute or hour.', $exceedLimit))
```